verification/cbmc/stubs/aws_hash_table_no_slots_override.c

```diff
@@ -65,7 +65,11 @@ void make_hash_table_with_no_backing_store(struct aws_hash_table *map, size_t ma
      * CBMC will detect any attempt to use the slots.  This ensures that no code will ever accidentally use the values
      * in the slots, ensuring maximal nondeterminism.
      */
+    if (map == NULL) {
+        return;
+    }
     map->p_impl = bounded_malloc(sizeof(struct hash_table_state));
+    __CPROVER_assume(map->impl);
     __CPROVER_assume(map->p_impl->entry_count <= max_table_entries);
```

Comment:
Perhaps,
```c
/**
 * Allocate a hash_table_state with no memory for the slots.
 * Since CBMC runs with memory safety assertions on,
 * CBMC will detect any attempt to use the slots.
 * This ensures that no code will ever accidentally use the values
 * in the slots, ensuring maximal nondeterminism.
 */
void make_hash_table_with_no_backing_store(struct aws_hash_table *map, size_t max_table_entries) {
    if (map != NULL) {
        map->p_impl = bounded_malloc(sizeof(struct hash_table_state));
        __CPROVER_assume(map->impl);
        __CPROVER_assume(map->p_impl->entry_count <= max_table_entries);
    }
}
```

Reply:
Yep this is nice! Updated to it